databricks_cli/stack/api.py

```diff
@@ -0,0 +1,340 @@
+                                RESOURCE_PHYSICAL_ID: physical_id,
+                                RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return resource_deploy_info
+
+    def validate_status(self, stack_status):
+        """
+        Validate fields within a stack status. This ensures that a stack status has the
+        necessary fields for stack deployment to function well.
+
+        If there is an error here, then it is either an implementation error that must be fixed by
+        a developer or the User edited the stack status file created by the program.
+
+        :param stack_status: dict- stack status that is created by the program.
+        :return: None. Raises errors to stop deployment if there is a problem.
+        """
+        if STACK_NAME not in stack_status:
+            raise StackError("'%s' not in status" % STACK_NAME)
+        if STACK_DEPLOYED not in stack_status:
+            raise StackError("'%s' not in status" % STACK_DEPLOYED)
+        for deployed_resource in stack_status[STACK_DEPLOYED]:
+            if RESOURCE_ID not in deployed_resource:
+                raise StackError("%s doesn't exist in deployed resource status" % RESOURCE_ID)
+            if RESOURCE_SERVICE not in deployed_resource:
+                raise StackError("%s doesn't exist in deployed resource status" % RESOURCE_SERVICE)
+            if RESOURCE_PHYSICAL_ID not in deployed_resource:
+                raise StackError("%s doesn't exist in deployed resource status" %
+                                 RESOURCE_PHYSICAL_ID)
+
+    def validate_config(self, stack_config):
+        """
+        Validate fields within a stack configuration. This ensures that an inputted configuration
+        has the necessary fields for stack deployment to function well.
+
+        :param stack_config: dict- stack config that is inputted by the user.
+        :return: None. Raises errors to stop deployment if there is a problem.
+        """
+        if STACK_NAME not in stack_config:
+            raise StackError("'%s' not in configuration" % STACK_NAME)
+        if STACK_RESOURCES not in stack_config:
+            raise StackError("'%s' not in configuration" % STACK_RESOURCES)
+        for resource in stack_config[STACK_RESOURCES]:
+            if RESOURCE_ID not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_ID)
+            if RESOURCE_SERVICE not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_SERVICE)
+            if RESOURCE_PROPERTIES not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_PROPERTIES)
+
+    def deploy_config(self, stack_config, stack_status=None):
+        self.validate_config(stack_config)
+        if stack_status:
+            self.validate_status(stack_status)
+            resource_to_deployed_resource = self._get_previous_resource_deploy_map(stack_status)
```

Comment:
`resource_to_deployed_resource` -> `resource_id_to_status`. Let's stick to the term `status` for deployed resource confs

Reply:
not sure if it should be resource_id_to_status since the key is both id and service

Reply:
`resource_to_status`?